yt/data_objects/selection_objects/data_selection_objects.py

```diff
@@ -72,12 +72,16 @@ def __init__(self, ds, field_parameters, data_source=None):
             self.field_parameters.update(data_source.field_parameters)
         self.quantities = DerivedQuantityCollection(self)
 
+    def _get_selector_class(self):
```

Comment:
note to reviewers: this change was needed so that the selector that is chosen can depend on the geometry. this method gets over-ridden in the new selector object.

Reply:
I don't remember if this was something I experimented with or not, but it seems like a good idea.  Although, I wonder if we could just wrap it in the `selector` property itself?